catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/QueryMetacardApplication.java

```diff
@@ -23,41 +23,88 @@
 import ddf.catalog.CatalogFramework;
 import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.impl.SortByImpl;
 import ddf.catalog.operation.CreateRequest;
 import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.QueryRequest;
 import ddf.catalog.operation.UpdateRequest;
 import ddf.catalog.operation.UpdateResponse;
 import ddf.catalog.operation.impl.CreateRequestImpl;
 import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
 import ddf.catalog.operation.impl.UpdateRequestImpl;
-import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import org.codice.ddf.catalog.ui.metacard.query.data.model.QueryBasic;
 import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortOrder;
 import spark.servlet.SparkApplication;
 
 public class QueryMetacardApplication implements SparkApplication {
 
   private static final Gson GSON =
       new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
 
+  private static final int MIN_START = 1;
+
+  private static final int MAX_PAGE_SIZE = 100;
+
+  private static final String START = "start";
+
+  private static final String COUNT = "count";
+
   private final CatalogFramework catalogFramework;
 
   private final EndpointUtil endpointUtil;
 
-  public QueryMetacardApplication(CatalogFramework catalogFramework, EndpointUtil endpointUtil) {
+  private final FilterBuilder filterBuilder;
+
+  public QueryMetacardApplication(
+      CatalogFramework catalogFramework, EndpointUtil endpointUtil, FilterBuilder filterBuilder) {
     this.catalogFramework = catalogFramework;
     this.endpointUtil = endpointUtil;
+    this.filterBuilder = filterBuilder;
   }
 
   @Override
   public void init() {
     get(
         "/queries",
         (req, res) -> {
-          List<Metacard> metacards = endpointUtil.getMetacardListByTag(QUERY_TAG);
-
-          return metacards.stream().map(QueryBasic::new).collect(Collectors.toList());
+          Map<String, String[]> params = req.queryMap().toMap();
+
+          int start = getOrDefaultParam(params, START, MIN_START);
+          int count = getOrDefaultParam(params, COUNT, MAX_PAGE_SIZE);
+
+          Filter filter = filterBuilder.attribute(Core.METACARD_TAGS).is().like().text(QUERY_TAG);
+
+          QueryRequest queryRequest =
+              new QueryRequestImpl(
+                  new QueryImpl(
+                      filter,
+                      start,
+                      count,
+                      new SortByImpl(Core.MODIFIED, SortOrder.DESCENDING),
+                      false,
+                      TimeUnit.SECONDS.toMillis(10)),
+                  false);
```

Comment:
This will work since we're under the internal page size of 1000, but if it (internal CatalogFramework page size) were smaller than the page size we're using here, it would not return all the results (since we're not using resultiterable).